sdk/typespec/typespec_client_core/src/http/context.rs

```diff
@@ -15,6 +15,16 @@ pub struct Context<'a> {
     type_map: Cow<'a, HashMap<TypeId, Arc<dyn Any + Send + Sync>>>,
 }
 
+impl Context<'static> {
+    /// Creates a new [`Context`] that borrows from the current [`Context`].
+    pub fn into_borrowed(&self) -> Context<'_> {
```

Comment:
This is a counterpart to `Context<'a>::into_owned()`. Since a `Context` holds a `Cow<'a, HashMap<...>>`, it is constrained to a specific lifetime. The `into_owned` and `into_borrowed` methods allow you to expand/contract that lifetime as necessary. The `into_owned` method clones the `HashMap` and converts it to a `Cow::Owned`, detaching it from the original `Context` that was provided. The `into_borrowed` method (which I now realize I should rename to `to_borrowed`, which I'll do in a minute) creates a new `Context<'a>` by *borrowing* the `HashMap`.

Reply:
@heaths this, and the one above in `PageStream`, are the only changes to core SDK components

Reply:
Given you commented on a previous commit and I was viewing all, this didn't show up for me to comment on, so see <https://github.com/Azure/azure-sdk-for-rust/pull/2577#discussion_r2080733536>. I'm fine with this in concept, but some suggestions/brainstorming on exactly impl.